### PR TITLE
feat(middleware): switch to auto-ensure package

### PR DESCRIPTION
**Pull request by RealShadowNova**

### CODE REVIEW

File: src/lib/structures/Josh.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export class Josh<StoredValue = unknown> {
 
     this.middlewares = new MiddlewareStore({ instance: this });
 
-    if (autoEnsure !== undefined) this.use(new AutoEnsure<StoredValue>(autoEnsure));
+    if (autoEnsure !== undefined) this.use(new AutoEnsure(autoEnsure) as unknown as Middleware<NonNullObject, StoredValue>);
```
Comment (dan-online) [inline src/lib/structures/Josh.ts:88]: Since it's been moved out of core, this doesn't need the "as unknown as Middleware"

**Review by dan-online — CHANGES_REQUESTED** [review]

File: src/lib/structures/Josh.ts (hunk 1 above)
Comment (RealShadowNova) [reply src/lib/structures/Josh.ts:88]: Not until after [josh-development/utilities#9](https://github.com/josh-development/utilities/pull/9) is merged. Which is why for now I'm going to mark this PR as blocked.

**Comment by RealShadowNova** [comment]
This PR is currently awaiting the merge of [josh-development/utilities#9](https://github.com/josh-development/utilities/pull/9)

**Comment by RealShadowNova** [comment]
This needs to be re-approached after providers have been finished.

**Pull request closed without merging**